### PR TITLE
Mkfs refactor

### PR DESCRIPTION
**Pull request by protoCall7**
Extracted and cleaned up mkfs logic from LVM.pm and Partitions.pm per issue #470 

Any input would be greatly appreciated.  I have not had a chance to spin up a test machine to try mkfs() on yet, however 'dzil test --no-author' is passing all tests.

### CODE REVIEW

**Comment by ferki** [comment]
Thanks for this! I only accidentally close-reopened it :) I'll do a proper review once I'm at my laptop!

**Comment by protoCall7** [comment]
Thanks @ferki!

If you don't have a handy machine to do proper testing with, I will be spinning up a few VM's to test on this evening, and can report my results here.

File: lib/Rex/Commands/Mkfs.pm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+=cut
+
+sub mkfs {
+  my ( $devname, %option ) = @_;
+
+  if ( exists $option{fstype} ) {
+    if ( $option{fstype} eq "swap" ) {
+      Rex::Logger::info("Creating swap space on /dev/$devname");
+      run "mkswap -f /dev/$devname";
+    }
```
Comment (ferki) [inline lib/Rex/Commands/Mkfs.pm]: I think it's not possible anymore to specify a label for a swap partition. Can you make sure this possibility is not lost?

**Comment by ferki** [comment]
@protoCall7: looks good, thanks! 

I'm not sure if the mkfs logic needs to have its own module, as it contains only a single function. I guess it can be kept in `Rex::Commands::Fs` (as in "that module contains filesystem related functions"), but it's not a very strong preference, just an idea. @krimdomu: do you have any preferences on this?

Apart from that, I left only one small comment about swap + labels. I think we can merge this once it is fixed + all commits squashed into one.

**Comment by protoCall7** [comment]
Updated the mkswap portion to support -L as well.

File: lib/Rex/Commands/Partition.pm [hunk 2]
```diff
@@ -225,42 +226,8 @@ sub partition {
...
-      $add_opts .= " -L $label ";
-    }
-
-    run "mkswap $add_opts /dev/$disk$part_num";
-  }
-  else {
-    die("Can't format partition with $option{fstype}");
-  }
-
+  mkfs "$disk$part_num";
```
Comment (ferki) [inline lib/Rex/Commands/Partition.pm]: How `Rex::Commands::Fs::mkfs` would know about the filesystem options (e.g. what file system type to create)? Would it still skip creating a file system if `non-fs`, `none` or `''` is passed as requested filetype?

**Comment by ferki** [comment]
ping:

>I'm not sure if the mkfs logic needs to have its own module, as it contains only a single function. I guess it can be kept in Rex::Commands::Fs (as in "that module contains filesystem related functions"), but it's not a very strong preference, just an idea.

@krimdomu: do you have any preferences on this?

ps.: if we're going to put it into a new module, we need to record it in `bin/rexify` and `docs/pp.args` too.

**Comment by krimdomu** [comment]
@ferki 
i think this is okay. Also it would be nice to have an "ensure" option, so that this function get idempotent. (checks partition with some command which fs it has, and if it is the requested one, do nothing)
But this can be done with a later code change.

**Comment by krimdomu** [comment]
Hi @protoCall7 

Thank you for this patch. I've just merged it into master branch.

File: lib/Rex/Commands/Partition.pm (hunk 2 above)
Comment (ferki) [inline lib/Rex/Commands/Partition.pm]: This PR looks like was merged for some reason without resolving this part of the problem, and lead us to #905 :(

**Pull request merged**